lopper/assists/baremetallinker_xlnx.py

```diff
@@ -30,19 +30,72 @@
 sys.path.append(os.path.dirname(__file__))
 from baremetalconfig_xlnx import scan_reg_size, get_cpu_node
 from bmcmake_metadata_xlnx import to_cmakelist
+from domain_access import *
 
 def is_compat( node, compat_string_to_test ):
     if re.search( "module,baremetallinker_xlnx", compat_string_to_test):
         return xlnx_generate_bm_linker
     return ""
 
+
 # tgt_node: is the baremetal config top level domain node number
 # sdt: is the system device-tree
 # options: baremetal application source path
 def get_memranges(tgt_node, sdt, options):
     root_node = sdt.tree[tgt_node]
     root_sub_nodes = root_node.subnodes()
     mem_nodes = []
+
+    domain_nodes = domain_get_subnodes(sdt.tree)
+    # Get the Baremetal Domain
+    for domain_node in domain_nodes:
+        bm_domain = []
+        if domain_node.propval('os,type') != ['']:
+            if re.search('baremetal', domain_node.propval('os,type', list)[0]):
+                    bm_domain.append(domain_node)
+        elif domain_node.propval('cpus') != ['']:
+                print('ERROR: os,type property is missing in the domain', domain_node.name)
+
+        match_cpunodes = get_cpu_node(sdt, options)
+
+        # Check whether domain is mapped for current processor or not
+        if bm_domain:
+            shared_mem = []
+            if not domain_node['cpus'].value[0] == match_cpunodes[0].parent.phandle:
```

Comment:
Minor comment: Similar to my other comment, this could just be domain_node['cpus'][0]